src/NzbDrone.Common/Http/HttpHeader.cs

```diff
@@ -4,37 +4,92 @@
 using System.Collections.Specialized;
 using System.Text;
 using NzbDrone.Common.Extensions;
+using System.Collections;
+using System.Globalization;
 
 namespace NzbDrone.Common.Http
 {
-    public class HttpHeader : Dictionary<string, object>
+    public class HttpHeader : NameValueCollection, IEnumerable<KeyValuePair<string, string>>, IEnumerable
```

Comment:
In some cases multiple headers are allowed. That's why the webrequestheadercollection uses namevaluecollection as well.